docs/features/logging.md

```diff
@@ -32,20 +42,123 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+By doing so, unhandled exceptions that might occur in other middleware components will also be logged by the `ExceptionHandlingMiddleware` component.
+
+## Logging incoming requests
+
+The `RequestTrackingMiddleware` class can be added to the <span>ASP.NET</span> Core pipeline to log all received HTTP requests.
+The incoming requests are logged by this middleware component using the `ILogger` implementations that are configured in the project.
+
+The HTTP request headers are logged by default, except certain security headers are by default omitted: `Authentication`, `X-Api-Key` and `X-ARR-ClientCert`.
+The HTTP request body is not logged by default.
+See [configuration](#configuration) for more details.
+
+**Example**
+
+`HTTP Request POST http://localhost:5000/weatherforecast completed with 200 in 00:00:00.0191554 at 03/23/2020 10:12:55 +00:00 - (Context: [Content-Type, application/json], [Body, {"today":"cloudy"}])`
+
+**Usage**
+
+To use this middleware, add the following line of code in the `Startup.Configure` method:
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEvironment env)
+{
+    app.UseRequestTracking();
+
+    ...
+    app.UseMvc();
+}
+```
+
+**Configuration**
+
+The request tracking middleware has several configuration options to manipulate what the request logging emits should contain.
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+{
+    app.UseRequestTracking(options =>
+    {
+        // Whether or not the HTTP request body should be included in the request tracking logging emits.
+        // (default: `false`)
+        options.IncludeRequestBody = true;
+
+        // Whether or not the configured HTTP request headers should be included in the request tracking logging emits.
+        // (default: `true`)
+        options.IncludeRequestHeaders = true;
+
+        // All omitted HTTP request header names that should always be excluded from the request tracking logging emits.
+        // (default: `[ "Authentication", "X-Api-Key", "X-ARR-ClientCert" ]`)
+        options.OmittedRequestHeaderNames.Add("X-My-Secret-Header");
+    });
+
+    ...
+    app.UseMvc();
+}
+```
+
+Optionally, one can inherit from this middleware component and override the default request header sanitization to run some custom functionality during the filtering.
+
+Following example shows how the request security headers can be emptied by not omitted:
+
+```csharp
+public class EmptyButNotOmitRequestTrackingMiddleware : RequestTrackingMiddleware
+{
+    public EmptyButNotOmitRequestTrackingMiddleware(
+        RequestDelegate next,
+        ILogger<RequestTrackingMiddleware> logger) 
+        : base(next, logger)
+        {
+        }
+
+    protected override IDictionary<string, StringValues> SanitizeRequestHeaders(IDictionary<string, StringValues> requestHeaders)
+    {
+        requestHeaders["X-Api-Key"] = String.Empty;
```

Comment:
Like this or just call remove on it? Anyhow, it's just a sample

Reply:
Well, the example here is to show that you make some custom logic during sanitization of the headers. If we would call remove, we could have done it in the options by calling `.OmittedRequestHeaderNames.Remove("X-Api-Key")`, but this example tries to show that you can run something else: not removing, but clearing the values.

But indeed, just for demonstration purposes. I thought that clearing values would also be a good thing to have bc you can see in the logs if a header was added without exposing secret information.

Reply:
Oh I get it now, thought it was to remove it!

Maybe this proves more what we want to show?

```suggestion
        requestHeaders["X-Api-Key"] = "<redacted>";
```